user/src/com/google/gwt/user/server/rpc/RemoteServiceServlet.java

```diff
@@ -94,6 +96,20 @@ static SerializationPolicy loadSerializationPolicy(HttpServlet servlet,
           try {
             serializationPolicy = SerializationPolicyLoader.loadFromStream(is,
                 null);
+            if (serializationPolicy.hasClientFields()) {
+              if (ENABLE_ENHANCED_CLASSES) {
+                servlet.log("WARNING: Enhanced JPA client fields are in use for this " +
+                        "application. See https://github.com/gwtproject/gwt/issues/9709 for " +
+                        "more detail. on the vulnerability that this presents.");
+              } else {
+                servlet.log("ERROR: Service uses enhanced classes, which are unsafe. Review " +
+                        "build logs to see where this can be fixed, or set " +
```

Comment:
Little more explicit:

`Service de-/serializes enhanced JPA/JDO classes which is unsafe. Review build logs to see which classes are affected, ..`